streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
I'm wondering if we can change the name of the method `StreamThread#isInitialized()` to something else a bit more meaningful and then that method could become:
`return state == State.RUNNING || state == State.DEAD`

We then don't need the `if(streamThread.state() == StreamThread.State.DEAD)` as `streamThread.tasks()` will be empty so we'll return an empty `List`